src/app/MessageDef/StatusIB.cpp

```diff
@@ -151,27 +152,10 @@ CHIP_ERROR StatusIB::ToChipError() const
 
 StatusIB::StatusIB(CHIP_ERROR aError)
 {
-    if (aError.IsPart(ChipError::SdkPart::kIMClusterStatus))
-    {
-        mStatus        = Status::Failure;
-        mClusterStatus = MakeOptional(aError.GetSdkCode());
-        return;
-    }
-
-    mClusterStatus = NullOptional;
-    if (aError == CHIP_NO_ERROR)
-    {
-        mStatus = Status::Success;
-        return;
-    }
-
-    if (aError.IsPart(ChipError::SdkPart::kIMGlobalStatus))
-    {
-        mStatus = static_cast<Status>(aError.GetSdkCode());
-        return;
-    }
+    ClusterStatusCode statusCode(aError);
```

Comment:
Why not just have this constructor delegate to the `ClusterStatusCode` constructor in the header?

As in, this whole constructor definition in the header should be:

```
StatusIB(CHIP_ERROR aError) : StatusIB(ClusterStatusCode(aError)) {}
```

Reply:
I tried to make it a member, but when we decode TLV, we decode directly into values (and those may not (yet?) be valid as a combination of status and code). Such a change has probably slightly more ripple effects.

Reply:
Updated code to place the definition in the header and re-use the clusterstatuscode constructor.